platform-tests/src/acceptance/compose_broker_test.go

```diff
@@ -1,22 +1,91 @@
 package acceptance_test
 
 import (
+	"fmt"
+	"io/ioutil"
+
 	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/generator"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/helpers"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gbytes"
 	. "github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("ComposeBroker", func() {
-
+var _ = Describe("Compose broker - MongoDB", func() {
 	const (
-		serviceName = "mongodb"
+		serviceName  = "mongodb"
+		testPlanName = "tiny"
 	)
 
 	It("is registered in the marketplace", func() {
-		marketplace := cf.Cf("marketplace").Wait(DEFAULT_TIMEOUT)
-		Expect(marketplace).To(Exit(0))
-		Expect(marketplace).To(Say(serviceName))
+		plans := cf.Cf("marketplace").Wait(DEFAULT_TIMEOUT)
+		Expect(plans).To(Exit(0))
+		Expect(plans).To(Say(serviceName))
+	})
+
+	It("has the expected plans available", func() {
+		plans := cf.Cf("marketplace", "-s", serviceName).Wait(DEFAULT_TIMEOUT)
+		Expect(plans).To(Exit(0))
+		Expect(plans.Out.Contents()).To(ContainSubstring("tiny"))
+	})
+
+	Context("creating a database instance", func() {
+		// Avoid creating additional tests in this block because this setup and
+		// teardown is slow (several minutes).
+
+		var (
+			appName        string
+			dbInstanceName string
+		)
+		BeforeEach(func() {
+			appName = generator.PrefixedRandomName("CATS-APP-")
+			dbInstanceName = generator.PrefixedRandomName("test-db-")
+			Expect(cf.Cf("create-service", serviceName, testPlanName, dbInstanceName).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			pollForServiceCreationCompletion(dbInstanceName)
+
+			fmt.Fprintf(GinkgoWriter, "Created MongoDB instance: %s\n", dbInstanceName)
+
+			Expect(cf.Cf(
+				"push", appName,
+				"--no-start",
+				"-b", config.GoBuildpackName,
+				"-p", "../../example-apps/healthcheck",
+				"-f", "../../example-apps/healthcheck/manifest.yml",
+				"-d", config.AppsDomain,
+			).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+
+			Expect(cf.Cf("bind-service", appName, dbInstanceName).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			Expect(cf.Cf("start", appName).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+		})
+
+		AfterEach(func() {
+			cf.Cf("delete", appName, "-f").Wait(DEFAULT_TIMEOUT)
+
+			Expect(cf.Cf("delete-service", dbInstanceName, "-f").Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			// Poll until destruction is complete, otherwise the org cleanup (in AfterSuite) fails.
+			pollForServiceDeletionCompletion(dbInstanceName)
+		})
+
+		It("binds a DB instance to mongodb-example-app that matches our criteria", func() {
```

Comment:
"mongodb-example-app" mentioned in the test title. Think this should now be "healthcheck app"